tooling-tests/src/test/java/io/ballerina/persist/tools/ToolingTestUtils.java

```diff
@@ -0,0 +1,95 @@
+import io.ballerina.projects.environment.Environment;
+import io.ballerina.projects.environment.EnvironmentBuilder;
+import org.testng.Assert;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * persist tool test Utils.
+ */
+public class ToolingTestUtils {
+
+    private static PrintStream errStream = System.err;
+
+    public static final String SAMPLES_DIRECTORY = "input";
+
+    public static final String REFERENCE_DIRECTORY = "output";
+    public static final String GENERATED_SOURCES_DIRECTORY = Paths.get("build", "generated-sources").toString();
+    public static final Path RESOURCE_PATH = Paths.get("src", "test", "resources", "test-src")
+            .toAbsolutePath();
+    private static final Path DISTRIBUTION_PATH = Paths.get(".." + File.separator, "target", "ballerina-runtime")
+            .toAbsolutePath();
+    private static ProjectEnvironmentBuilder getEnvironmentBuilder() {
+        Environment environment = EnvironmentBuilder.getBuilder().setBallerinaHome(DISTRIBUTION_PATH).build();
+        return ProjectEnvironmentBuilder.getBuilder(environment);
+    }
+
+    public static void assertGeneratedSources(String subDir, String configFile) {
+        Path sourceDirPath = Paths.get(GENERATED_SOURCES_DIRECTORY, subDir);
+        Path referenceFilePath = Paths.get(RESOURCE_PATH.toString(), REFERENCE_DIRECTORY, subDir);
+        Path actualConfigFilePath = sourceDirPath.resolve(configFile);
+        Path referenceConfigFilePath = referenceFilePath.resolve(configFile);
+        generateSourceCode(sourceDirPath.toAbsolutePath());
+        Assert.assertTrue(Files.exists(actualConfigFilePath));
+        Assert.assertEquals(readContent(actualConfigFilePath), readContent(referenceConfigFilePath));
+    }
+
+    public static void assertGeneratedSourcesNegative(String subDir, String configFile) {
+        Path sourceDirPath = Paths.get(GENERATED_SOURCES_DIRECTORY, subDir);
+        Path actualConfigFilePath = sourceDirPath.resolve(configFile);
+        generateSourceCode(sourceDirPath);
+        Assert.assertFalse(Files.exists(actualConfigFilePath));
+    }
+
+    private static void generateSourceCode(Path sourcePath) {
+        Class<?> persistInitClass;
+        try {
+            persistInitClass = Class.forName("io.ballerina.persist.cmd.Init");
+            Init persistCmdInit = (Init) persistInitClass.getDeclaredConstructor().newInstance();
+            persistCmdInit.setSourcePath(sourcePath.toAbsolutePath().toString());
+            persistCmdInit.setEnvironmentBuilder(getEnvironmentBuilder());
+            persistCmdInit.execute();
+        } catch (ClassNotFoundException | IllegalAccessException | InstantiationException |
+                NoSuchMethodException | InvocationTargetException e) {
+            errStream.println(e.getMessage());
```

Comment:
```suggestion
            throw new RuntimeException(e);
```

Reply:
In the code review, suggestion was to print the error without creating a runtime error right?